### PR TITLE
serialize asset/dataset timetable conditions in OpenLineage info also for older supported Airflow 2 versions

### PR DESCRIPTION
**Pull request by mobuchowski**
Currently, that information is included in OpenLineage events only for Airflow 2.10 and Airflow 3. 
We want to include it in OL events when running on Airflow 2.8 and 2.9 too.

### CODE REVIEW

File: providers/src/airflow/providers/openlineage/utils/utils.py [hunk 1]
```diff
@@ -262,9 +262,31 @@ class DagInfo(InfoJsonEncodable):
         "start_date",
         "tags",
     ]
-    casts = {"timetable": lambda dag: dag.timetable.serialize() if getattr(dag, "timetable", None) else None}
+    casts = {"timetable": lambda dag: DagInfo.serialize_timetable(dag)}
     renames = {"_dag_id": "dag_id"}
 
+    @classmethod
+    def serialize_timetable(cls, dag):
```
Comment (Lee-W) [inline providers/src/airflow/providers/openlineage/utils/utils.py]:
```suggestion
    def serialize_timetable(cls, dag: DAG) -> dict[str, Any]:
```

File: providers/src/airflow/providers/openlineage/utils/utils.py [hunk 2]
```diff
@@ -262,9 +262,31 @@ class DagInfo(InfoJsonEncodable):
...
+        serialized = dag.timetable.serialize()
+        if serialized != {} and serialized is not None:
+            return serialized
+        if (
+            hasattr(dag, "dataset_triggers")
+            and isinstance(dag.dataset_triggers, list)
+            and len(dag.dataset_triggers) > 0
```
Comment (Lee-W) [inline providers/src/airflow/providers/openlineage/utils/utils.py]:
```suggestion
            and len(dag.dataset_triggers)
```

File: providers/src/airflow/providers/openlineage/utils/utils.py [hunk 3]
```diff
@@ -262,9 +262,31 @@ class DagInfo(InfoJsonEncodable):
...
+        ):
+            triggers = dag.dataset_triggers
+            return {
+                "dataset_condition": {
```
Comment (Lee-W) [inline providers/src/airflow/providers/openlineage/utils/utils.py:280]: I think we probably would need to make it asset compatible?

Comment (Lee-W) [reply providers/src/airflow/providers/openlineage/utils/utils.py:280]: or is it used only in 2.8 and 2.9

Comment (mobuchowski) [reply providers/src/airflow/providers/openlineage/utils/utils.py:280]: @Lee-W Yes, this is only for 2.8 and 2.9.

**Review by Lee-W — APPROVED** [review]

**Pull request merged**